### PR TITLE
Prevents abuse of the BEPIS to syphon cash from cargo.

### LINKED ISSUE #52705: Station budget can be drained directly via using B.E.P.I.S and a Departmental Card (Cargo Budget).
143479 Round ID:

## Reproduction:
Station budget can be drained directly via using B.E.P.I.S and a Departmental Card (Cargo Budget). Click on B.E.P.I.S with the departmental card, then you can deposit as much credits as the station has. Afterwards click to withdraw the credits, this will come out as a credit chip, which can then be put onto any ID. This allows you to instantly drain the stations funds without alerting anyone.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

Yeah I didn't really get around to this until now, I apologize.
New solution:
**The BEPIS no longer stores an internal quantity of credits inside of itself before starting, with the chance of success being determined by the attempted amount of credits being deposited. Once deposited, the machine automatically activates, spitting out rewards/failure alerts.**
**This is also reflected in the required UI changes to make this simplification work, bare with me as I will probably need to tweak a bit more to get it perfect, but it's easily going in a more reasonable direction.**
_Plus, this means that the BEPIS doesn't need to be manually swiped, instead autograbbing the target's active ID! Manual swiping can still be used to overwrite the old value._

## Why It's Good For The Game

Fixes #52705, and prevents cheesing of the core mechanic of the BEPIS.

## Changelog
:cl:
fix: You can no longer use the BEPIS to rob the cargo budget blind. It now auto-starts depending on the attempted deposited credits.
/:cl:

### CODE REVIEW

**Comment by coiax** [comment]
Um, I thought being able to use the cargo budget card on the BEPIS was a feature.

Since Cargo's the only one able to get the sums of money required to pay for BEPIS tech, in my experience.

**Comment by ATH1909** [comment]
:noooooo:

**Comment by ATH1909** [comment]
> Um, I thought being able to use the cargo budget card on the BEPIS was a feature.
> 
> Since Cargo's the only one able to get the sums of money required to pay for BEPIS tech, in my experience.

maybe we could change how the bepis works so that it just spends the money directly from the account used, instead of moving it into the bepis first (which is what lets us move money from the cargo budget into the bepis and then into the real world)?

if you do want to patch this, could you please make withdrawing money from the vault not take 10,000,000,000 years?

**Comment by JohnFulpWillard** [comment]
I thought using the Cargo departmental budget was the sole reason the Bepis was moved from Science to Cargo? I think a better alternative would be to remove the withdrawal funds button, money put in the bepis should be in the bepis for good.

File: code/modules/research/bepis.dm [hunk 1]
```diff
@@ -61,6 +61,9 @@
 		return
 	if(istype(O, /obj/item/card/id))
 		var/obj/item/card/id/Card = O
+		if(istype(Card, /obj/item/card/id/departmental_budget)
```
Comment (Mothblocks) [inline code/modules/research/bepis.dm:64]:
```suggestion
		if(istype(Card, /obj/item/card/id/departmental_budget))
```

Sir

Comment (ArcaneMusic) [reply code/modules/research/bepis.dm:64]: Oh I've had the fix as an uncommited change for about 7 hours now I'm just mulling about doing the rest of the PR better.

**Comment by moocowswag** [comment]
This really hinders department cooperation and further incentivizes breaking into the vault and siphoning after making a ton of money with a research bomb.

If scientists could claim atleast some of the money they earned from research bombs it would be a really good compromise to this change.

Because as of right now scientists have no incentive to make 50ks due to it no longer earning research points, just money that goes to the station's budget.

**Comment by ArcaneMusic** [comment]
So I sat down and worked with @ATH1909 's soulution over the weekend, check the description for updated solution, in bold.

**Comment by ArcaneMusic** [comment]
https://user-images.githubusercontent.com/41715314/111389695-ca705b00-8687-11eb-811b-2a4030f62527.mp4

Demo of the above, undrafting.

File: code/modules/research/bepis.dm [hunk 2]
```diff
@@ -190,11 +176,14 @@
 		ui = new(user, src, "Bepis", name)
 		ui.open()
 	RefreshParts()
+	if(isliving(user))
+		var/mob/living/customer = user
+		account = customer?.get_bank_account()
```
Comment (Mothblocks) [inline code/modules/research/bepis.dm:181]:
```suggestion
		account = customer.get_bank_account()
```

It should be impossible for `customer` to be null after your `isliving` check.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Mothblocks** [comment]
One minor nitpick.

**Pull request merged**